support/scheduler-client/client_test.go

```diff
@@ -121,12 +121,190 @@ func TestSendSchedule(t *testing.T) {
 		RunOnce:   TestScheduleRunOnce,
 	}
 
-	error := scheduleClient.SendSchedule(schedule)
+	error := scheduleClient.AddSchedule(schedule)
 	if error != nil {
 		t.Error(error)
 	}
 }
 
+// Test method : QuerySchedule
+func TestQuerySchedule(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "GET" {
+			t.Errorf(TestUnexpectedMsgFormatStr, r.Method, "GET")
+		}
+
+		urlWithIdPath := ScheduleApiPath + "/" + TestScheduleId
+
+		if r.URL.EscapedPath() != urlWithIdPath {
+			t.Errorf(TestUnexpectedMsgFormatStr, r.URL.EscapedPath(), urlWithIdPath)
+		}
+
+		id := strings.TrimPrefix(r.URL.EscapedPath(), ScheduleApiPath+"/")
+
+		if id != TestScheduleId {
+			t.Errorf(TestUnexpectedMsgFormatStr, id, TestScheduleId)
+		}
+
+		schedule := models.Schedule{
+			Name:      TestScheduleName,
+			Start:     TestScheduleStart,
+			End:       TestScheduleEnd,
+			Frequency: TestScheduleFrequency,
+			Cron:      TestScheduleCron,
+			RunOnce:   TestScheduleRunOnce,
+		}
+
+		w.WriteHeader(http.StatusOK)
+
+		jsonBytes, err := schedule.MarshalJSON()
+		if err != nil {
+			t.Error(err.Error())
+		}
+
+		w.WriteHeader(http.StatusOK)
+		w.Write(jsonBytes)
+	}))
+
+	defer ts.Close()
+
+	u, err := url.Parse(ts.URL)
+	if err != nil {
+		t.Error(err.Error())
+	}
+
+	h := strings.Split(u.Host, ":")
+
+	intPort, e := strconv.Atoi(h[1])
+	if e != nil {
+		t.Error(e)
+	}
+
+	scheduleClient := SchedulerClient{
+		SchedulerServiceHost: h[0],
+		SchedulerServicePort: intPort,
+		OwningService:        "notifications",
+	}
+
+	receivedSchedule, err := scheduleClient.QuerySchedule(TestScheduleId)
+	if err != nil {
+		t.Error(err.Error())
+	}
+
+	if receivedSchedule.Name != TestScheduleName {
+		t.Errorf(TestUnexpectedMsgFormatStr, receivedSchedule.Name, TestScheduleName)
+	}
+
+	if receivedSchedule.Start != TestScheduleStart {
+		t.Errorf(TestUnexpectedMsgFormatStr, receivedSchedule.Start, TestScheduleStart)
+	}
+
+	if receivedSchedule.End != TestScheduleEnd {
+		t.Errorf(TestUnexpectedMsgFormatStr, receivedSchedule.End, TestScheduleEnd)
+	}
+
+	if receivedSchedule.Frequency != TestScheduleFrequency {
+		t.Errorf(TestUnexpectedMsgFormatStr, receivedSchedule.Frequency, TestScheduleFrequency)
+	}
+
+	if receivedSchedule.Cron != TestScheduleCron {
+		t.Errorf(TestUnexpectedMsgFormatStr, receivedSchedule.Cron, TestScheduleCron)
+	}
+
+	if receivedSchedule.RunOnce != TestScheduleRunOnce {
+		t.Errorf(TestUnexpectedMsgFormatStrForBoolVal, receivedSchedule.RunOnce, TestScheduleRunOnce)
+	}
+}
+
+// Test method : QueryScheduleWithName
+func TestQueryScheduleWithName(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "GET" {
```

Comment:
Do not hardcode the method names. Use package-defined constants: https://golang.org/pkg/net/http/#pkg-constants